data_structures/src/utxo_pool.rs

```diff
@@ -147,6 +150,53 @@ impl OwnUnspentOutputsPool {
     }
 }
 
+/// Struct that keeps the unspent outputs pool and the own unspent outputs pool
+#[derive(Debug)]
+pub struct NodeUtxos<'a> {
+    pub all_utxos: &'a UnspentOutputsPool,
+    pub own_utxos: &'a mut OwnUnspentOutputsPool,
```

Comment:
Maybe I would comment them: the `OwnUnspentOutputsPool` field has only the output pointers and not value transfer information (that's why you need the `UnspentOutputsPool`).